extensions/amp-story/1.0/amp-story.js

```diff
@@ -861,35 +859,23 @@ export class AmpStory extends AMP.BaseElement {
 
     const oldPage = this.activePage_;
 
-    // TODO(cvializ): Move this to the page class?
-    const activePriorSibling = targetPage.element.previousElementSibling;
-    const previousActivePriorSibling = this.element.querySelector(
-        `[${escapeCssSelectorIdent(PRE_ACTIVE_PAGE_ATTRIBUTE_NAME)}]`);
-
     this.activePage_ = targetPage;
 
     this.systemLayer_.resetDeveloperLogs();
     this.systemLayer_.setDeveloperLogContextString(
         this.activePage_.element.id);
 
+    if (oldPage) {
+      oldPage.setActive(false);
```

Comment:
I think we should disable the old page only when the next page is ready to be displayed, after this call to ``beforeVisible``. If this ``beforeVisible`` method takes time for some reason (low perf device or anything else), the previous page being unactive will make the UI look like it froze.

Reply:
moved. Originally changed because of race between `replay_` and `switchTo_` but now written so that `replay_` waits for `switchTo_` promise.